src/commands/Minion/slayershop.ts

```diff
@@ -0,0 +1,307 @@
+import { requiresMinion } from '../../lib/minions/decorators';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { SlayerRewardsShop } from '../../lib/slayer/slayerUnlocks';
+import { getSlayerReward } from '../../lib/slayer/slayerUtil';
+import { BotCommand } from '../../lib/structures/BotCommand';
+import { stringMatches, toTitleCase } from '../../lib/util';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			usage: '[unlock|lock|buy] [name:...string]',
+			usageDelim: ' ',
+			oneAtTime: true,
+			cooldown: 5,
+			altProtection: true,
+			subcommands: true,
+			aliases: ['sls', 'slshop'],
+			description: `Allows a player to purchase slayer unlocks.`,
+			examples: ['+sls unlock malevolent masqureade', '+slshop unlock slayer helmet'],
+			categoryFlags: ['minion']
+		});
+	}
+
+	@requiresMinion
+	async run(msg: KlasaMessage, [_input]: [string]) {
+		if (
+			msg.flagArgs.unlocks ||
+			msg.flagArgs.help ||
+			_input === 'help' ||
+			_input === 'unlocks'
+		) {
+			// const myUnlocks = SlayerRewardsShop.filter(srs => { return srs.item !== undefined });
+			const unlockTable = table([
+				['Slayer Points', 'Name', 'Description', 'Type'],
+				...SlayerRewardsShop.map(i => [
+					i.slayerPointCost,
+					i.name,
+					i.desc,
+					i.item !== undefined
+						? 'item/buy'
+						: i.extendMult !== undefined
+						? 'extend'
+						: 'unlock'
+				])
+			]);
+			return msg.channel.sendFile(Buffer.from(unlockTable), `slayerRewardsItems.txt`);
+		}
+
+		let unlocks: string[] = [];
+		const myUnlocks = await msg.author.settings.get(UserSettings.Slayer.SlayerUnlocks);
+		const myPoints = await msg.author.settings.get(UserSettings.Slayer.SlayerPoints);
+
+		myUnlocks.forEach(u => {
+			unlocks.push(getSlayerReward(u));
+		});
+		const defaultMsg = `Current points: ${myPoints}\nYou currently have the following ` +
+			`rewards unlocked:\n\`${unlocks.join(`\n`)}\`\n\n` +
+			`Usage:\n\`${msg.cmdPrefix}slayershop [unlock|lock|buy] Reward\`\nExample:` +
+			`\n\`${msg.cmdPrefix}slayershop unlock Malevolent Masquerade\``
+		if (defaultMsg.length > 2000) {
+			return msg.channel.sendFile(Buffer.from(defaultMsg), `currentUnlocks.txt`);
+		} else {
+			return msg.channel.send(defaultMsg);
+		}
+	}
+
+	async buy(msg: KlasaMessage, [buyableName = '']: [string]) {
+		if (msg.flagArgs.items || msg.flagArgs.help || buyableName === 'help') {
+			const myUnlocks = SlayerRewardsShop.filter(srs => {
+				return srs.item !== undefined;
+			});
+			const unlockTable = table([
+				['Slayer Points', 'Name', 'Description', 'Type'],
+				...myUnlocks.map(i => [i.slayerPointCost, i.name, i.desc, 'item/buy'])
+			]);
+			return msg.channel.sendFile(Buffer.from(unlockTable), `slayerRewardsItems.txt`);
+		}
+
+		if (buyableName === '') {
+			throw `You must specify an item to purchase.\nTry:\n\`${msg.cmdPrefix}sls buy --help\``;
```

Comment:
we no longer throw strings like this in commands, you should do `return msg.channel.send()`